File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        vector<ListNode*> store;
        while(headB){
            store.push_back(headB);
            headB = headB->next;
        }
        while(headA){
           if (find(store.begin(), store.end(), headA) != store.end())
{
                return headA;
            }
            headA = headA->next;
        }
        return nullptr;
    }
};

File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    let arr = [];
    while(headB){
        arr.push(headB);
        headB = headB.next;
    }
    while(headA){
        if(arr.includes(headA)) return headA;
        headA = headA.next;
    }
    return null;
};